File: Laya/GPUSkinning/src/GPUSkinning/Datas/GPUSkinningAnimEvent.ts

/** 烘焙动画--动画剪辑事件 */
export default class GPUSkinningAnimEvent  
{
    /** 帧索引 */
    frameIndex:int = 0;

    /** 事件ID */
    eventId:int = 0;

    /** 比较 */
    CompareTo(other: GPUSkinningAnimEvent): int
    {
        return this.frameIndex > other.frameIndex ? -1 : 1;
    }

    
    FromBytes(data: ArrayBuffer): void
    {
        var b:Laya.Byte = new Laya.Byte(data);
        this.frameIndex = b.readInt32();
        this.eventId = b.readInt32();

    }

    static CreateFromBytes(data: ArrayBuffer):GPUSkinningAnimEvent
    {
        var obj = new GPUSkinningAnimEvent();
        obj.FromBytes(data);
        return obj;
    }
}

File: Unity3D/GPUSkinning/Assets/GPUSkinning/Scripts/GPUSkinningAnimEvent.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.IO;
using UnityEngine;

[System.Serializable]
public class GPUSkinningAnimEvent : System.IComparable<GPUSkinningAnimEvent>
{
    public int frameIndex = 0;

    public int eventId = 0;

    public int CompareTo(GPUSkinningAnimEvent other)
    {
        return frameIndex > other.frameIndex ? -1 : 1;
    }


    public static GPUSkinningAnimEvent CreateFromBytes(byte[] bytes)
    {
        GPUSkinningAnimEvent obj = new GPUSkinningAnimEvent();
        obj.FromBytes(bytes);
        return obj;
    }

    public void FromBytes(byte[] bytes)
    {

    }


    public MemoryStream ToSteam()
    {
        MemoryStream stream = new MemoryStream();
        BinaryWriter b = new BinaryWriter(stream);
        b.Write((int) frameIndex);
        b.Write((int) eventId);
        return stream;
    }
}
